site/_posts/2019-09-20-Velero-v1-1-on-vSphere.md

```diff
@@ -0,0 +1,253 @@
+---
+title: Velero v1.1 backing up and restoring apps on vSphere
+# image: https://placehold.it/200x200
+excerpt: Velero is an open source tool to safely back up, recover, and migrate Kubernetes clusters and persistent volumes. It works both on premises and in a public cloud. This post demonstrates how it can be used on vSphere.
+author_name: Cormac Hogan
+# author_avatar: https://placehold.it/64x64
+categories: ['kubernetes']
+# Tag should match author to drive author pages
+tags: ['Velero']
+---
+Velero version 1.1 provides support to backup Kubernetes applications deployed on vSphere. This post will provide detailed information on how to deploy and configure Velero to backup and restore a stateless application (`nginx`) that is running in Kubernetes on vSphere. At this time there is no vSphere plugin for snapshotting stateful applications on vSphere during a Velero backup. In this case, we rely on a third party program called `restic`. However this post does not include an example of how to backup a stateful application. That is available in another tutorial which can be found [here](./2019-20-09-Velero-v1-1-Stateful-Backup-vSphere.md).
+
+## Overview of steps
+
+* Download and extract Velero v1.1
+* Deploy and Configure Minio Object store
+* Install Velero using the `velero install` command, ensuring that both `restic` support and a Minio `publicUrl` are included
+* Run a test backup/restore of a stateless application that has been deployed on upstream Kubernetes
+
+## What this post does not show
+
+* How to install Velero v1.1 with PKS, Pivotal Container Service, on vSphere
+* A demonstration on how to do backup/restore of a stateful application (i.e. PVs)
+* The assumption is that the Kubernetes nodes in your cluster have internet access. This guide does not show how to use a local repository for Velero images
+
+## Download and extract Velero v1.1
+
+The [Velero v1.1 image can be found here](https://github.com/heptio/velero/releases/tag/v1.1.0.). Download and extract it to the desktop where you wish to manage your Velero backups, then copy or move the `velero` binary to somewhere in your $PATH.
+
+## Deploy and Configure Minio Object Store as a backup destination
+
+Velero sends metadata about the objects being backed up to an S3 Object Store. If you do not have an S3 Object Store available, Velero provides the manifest file to create a Minio S3 Object Store on your Kubernetes cluster. This means that all Velero backsup can be kept on-premises.
```

Comment:
Besides metadata, we also send the actual contents of the k8s resources. Might be useful to extend the explanation.

Reply:
Done